src/utils.rs

```diff
@@ -114,14 +114,11 @@ pub fn field_element_to_felt(felt: &FieldElement) -> Felt {
 
 pub fn to_state_diff_storage_mapping(
     storage_writes: HashMap<StorageEntry, Felt>,
-) -> Result<HashMap<Felt, HashMap<[u8; 32], Address>>, StateError> {
+) -> HashMap<Felt, HashMap<[u8; 32], Address>> {
     let mut storage_updates: HashMap<Felt, HashMap<[u8; 32], Address>> = HashMap::new();
     for ((address, key), value) in storage_writes {
         if storage_updates.contains_key(&address.0) {
```

Comment:
What about replacing the `if-else` with this? I think it should be equivalent, while also avoiding `.unwrap()`.
```suggestion
        let mut map = storage_updates.get(&address.0).cloned().unwrap_or_default();
        map.insert(key, Address(value));
        storage_updates.insert(address.0, map);
```

Reply:
yes! i really like this refactor, thank you for pointing it out.